core/src/main/java/utilcalc/core/parser/Parser.java

```diff
@@ -49,12 +65,42 @@ public static ReportInputs parse(String input) {
             LocalDate reportDate = requireLocalDate(general, REPORT_DATE);
             List<String> sources = optionalList(general, SOURCES, String.class);
 
+            List<SectionInputs> sections = new ArrayList<>();
+            optionalDepositsSection(parseResult).ifPresent(sections::add);
+
             return new ReportInputs(
-                    startDate, endDate, tenant, owner, reportPlace, reportDate, sources, List.of());
+                    startDate,
+                    endDate,
+                    tenant,
+                    owner,
+                    reportPlace,
+                    reportDate,
+                    sources,
+                    List.copyOf(sections));
         } catch (TomlInvalidTypeException e) {
             throw new ParsingException("Invalid data type: " + e.getMessage(), e);
         } catch (IllegalArgumentException e) {
             throw new ParsingException("Unknown error when parsing: " + e.getMessage(), e);
         }
     }
+
+    private static Optional<SectionInputs> optionalDepositsSection(TomlTable parseResult) {
+        if (!parseResult.contains(DEPOSITS_SECTION_NAME)) {
+            return Optional.empty();
+        }
+        TomlArray depositPayments = requireArray(parseResult, DEPOSITS_SECTION_NAME);
+        List<Payment> payments = new ArrayList<>(depositPayments.size());
+
+        for (int i = 0; i < depositPayments.size(); i++) {
+            TomlTable payment = depositPayments.getTable(i);
+            checkThatSectionContainsOnlyKnownFields(
```

Comment:
Telo for cyklu bych extrahoval do `parsePayment`, pak by sel cely for cyklus nahradit streamem bez explicitniho vytvareni listu a volani `List.add`